### ISSUE TITLE
Check failed: max_num_elems > 0 (0 vs. 0) 

### ISSUE THREAD

**Issue by paudom** [nerfstudio-project/nerfstudio#834]
**Describe the bug**
I'm trying to execute the preprocessing command on the nerfstudio documentation:
```bash
ns-process-data video --data {data_path} --output-dir {output_path} --no_gpu
```
> Using `no_gpu` flag as I do not have a display. Running these on a EC2 instance. AMD64, 20.04 Ubuntu with CUDA 11.6

But the following error appears:
```
[12:53:19] 🎉 Done converting video to images.                                                       process_data.py:178
[12:53:47] 🎉 Done downscaling images.                                                               process_data.py:322
           🎉 Done extracting COLMAP features.                                                       process_data.py:363
──────────────────────────────────────────────  💀 💀 💀 ERROR 💀 💀 💀  ───────────────────────────────────────────────
Error running command: colmap vocab_tree_matcher --database_path /home/ubuntu/Datasets/Processed/colmap/database.db 
--SiftMatching.use_gpu 0 --VocabTreeMatching.vocab_tree_path /home/ubuntu/.local/share/nerfstudio/vocab_tree.fbow
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
F20221025 12:53:48.190236 134291 cache.h:132] Check failed: max_num_elems > 0 (0 vs. 0) 
*** Check failure stack trace: ***
    @     0x55602e3a7a00  google::LogMessage::SendToLog()
    @     0x55602e3a3f46  google::LogMessage::Flush()
    @     0x55602e3a819f  google::LogMessageFatal::~LogMessageFatal()
    @     0x55602db72bbd  colmap::LRUCache<>::LRUCache()
    @     0x55602db61dd7  colmap::FeatureMatcherCache::Setup()
    @     0x55602db6de96  colmap::VocabTreeFeatureMatcher::Run()
    @     0x55602dcc7ad1  colmap::Thread::RunFunc()
    @     0x7f0c8a74cde4  (unknown)
    @     0x7f0c8a86a609  start_thread
    @     0x7f0c8a544133  clone
    @              (nil)  (unknown)
Aborted (core dumped)
```

**Reproduce**
Installed colmap using the `vcpkg` succesfully. Then, run the command written above to preprocess custom data.

**Expected behavior*
I expected colmap to run the preprocessing correctlY.

**Comment by tancik** [comment]
This error typically arrises when COLMAP fails to find poses. It can be cause by a number of issues such as too much motion blur, not enough images, or too many dynamic objects in the scene.

**Comment by paudom** [comment]
Hi @tancik do you know if nerfstudio has any video without any preprocessing to do the test to make sure that the problem is the video i'm using?
Thanks.

**Comment by tancik** [comment]
Yep, try giving this video a go - https://drive.google.com/file/d/19zEm-CpOjpVObFzGqrxou5ndBjb_fAui/view?usp=sharing

**Comment by tancik** [comment]
I guess it could be a problem with the libfreeimage version...? Out of curiosity, what is currently in your `output_path` directory after running the script?

**Comment by paudom** [comment]
My `output_path` is an empty folder created by me, with nothing in it. Once I execute the command `ns-process-data` the folder fills with the following folders/data:
- colmap/ --> where the database.db is created
- images/ --> I understand that is the place where the images are saved from each frame
- images_2/
- ....

**Comment by tancik** [comment]
Ok, that seems to be working as expected. I just wanted to make sure that images existed in the images folder. I'm not really sure how to help at this point, it seems like the issue exists with colmap.

**Comment by paudom** [comment]
Thanks a lot for your time! I'm closing for now, reopen if someone has the same issue!

**Comment by paudom** [comment]
Just to clarify, I decided to install COLMAP using the source. And using the same video and the one @tancik provided this error disappeared. It worked fine. I don't know what happened during the vcpkg installation.

**Comment by corbt** [comment]
I ran into this issue yesterday as well when running a working `nerfstudio` Docker image on a new host with a different GPU.

For future readers, it's worth clarifying that this issue is a generic problem that means that COLMAP wasn't able to place any images. The actual reason COLMAP failed can vary. You can sometimes get more information by manually running the COLMAP feature extractor from the command line manually and checking the output for errors.  In my case I ran `colmap feature_extractor --database_path test.db --image_path ./images --ImageReader.single_camera 1 --ImageReader.camera_model OPENCV  --SiftExtraction.use_gpu 0`, and saw that COLMAP wasn't able to access my GPU for some reason.

**Comment by liuyi0316** [comment]
hello, do you figure it out?

**Comment by fducau** [comment]
I made it work by converting all my `png` images to `jpg`  as suggested here: https://github.com/colmap/colmap/issues/1195#issuecomment-1718730025 

Apparently it could be related to how `freeimage` is compiled.
Going with this temporary solution for now.
